rusk-wallet/src/wallet/file.rs

```diff
@@ -4,107 +4,155 @@
 //
 // Copyright (c) DUSK NETWORK. All rights reserved.
 
-use std::fmt;
+use std::fmt::{Debug, Display};
+use std::hash::Hash;
 use std::path::{Path, PathBuf};
 use std::str::FromStr;
 
-/// Provides access to a secure wallet file
-pub trait SecureWalletFile {
-    /// Returns the path
-    fn path(&self) -> &WalletPath;
-    /// Returns the hashed password
-    fn pwd(&self) -> &[u8];
+use crate::dat::DatFileVersion;
+use crate::{Error, SecureWalletFile};
+
+use super::file_service::WalletFilePath;
+
+/// Wallet file structure that contains the path of the wallet file, the hashed
+/// password, and the file version
+#[derive(Debug, Clone)]
+pub struct WalletFile {
+    path: WalletPath,
+    pwd: Vec<u8>,
+    file_version: DatFileVersion,
+}
+
+impl SecureWalletFile for WalletFile {
+    type PathBufWrapper = WalletPath;
+
+    fn path(&self) -> &WalletPath {
+        &self.path
+    }
+
+    fn path_mut(&mut self) -> &mut WalletPath {
+        &mut self.path
+    }
+
+    fn pwd(&self) -> &[u8] {
+        &self.pwd
+    }
+
+    fn version(&self) -> DatFileVersion {
+        self.file_version
+    }
+}
+
+impl WalletFile {
+    /// Create a new wallet file
+    pub fn new(
+        path: WalletPath,
+        pwd: Vec<u8>,
+        file_version: DatFileVersion,
+    ) -> Self {
+        Self {
+            path,
+            pwd,
+            file_version,
+        }
+    }
 }
 
 /// Wrapper around `PathBuf` for wallet paths
 #[derive(PartialEq, Eq, Hash, Debug, Clone)]
 pub struct WalletPath {
-    /// Path of the wallet file
-    pub wallet: PathBuf,
+    /// Path to the wallet file
+    wallet: PathBuf,
     /// Directory of the profile
-    pub profile_dir: PathBuf,
+    profile_dir: PathBuf,
     /// Name of the network
-    pub network: Option<String>,
+    network: Option<String>,
```

Comment:
Why do we have network in the wallet path? It seems irrelevant to have it here, we have the `settings.rs` file in the binary folder (which should be moved to the library IMO) which has the network there, I don't think we should duplicate that

Reply:
I didn't touch the `WalletPath` shape and it had the `network` property before refactoring.